bigquery_etl/cli/query.py

```diff
@@ -1349,94 +1355,113 @@ def initialize(ctx, name, sql_dir, project_id, dry_run, parallelism):
         )
         sys.exit(1)
 
-    for query_file in query_files:
+    def _initialize(query_file):
+        client = bigquery.Client()
+        project, dataset, destination_table = extract_from_query_path(query_file)
+        full_table_id = f"{project}.{dataset}.{destination_table}"
+        table = None
+
         sql_content = query_file.read_text()
+        init_files = list(Path(query_file.parent).rglob("init.sql"))
+
+        # check if the provided file can be initialized and whether existing ones should be skipped
+        if "is_init()" in sql_content or len(init_files) > 0:
+            if skip_existing:
+                try:
+                    table = client.get_table(full_table_id)
+                    # table exists; skip initialization
+                    return
+                except NotFound:
+                    # continue with creating the table
+                    pass
```

Comment:
To allow that specifically for materialized views we could check if the table's `table_type` is "MATERIALIZED VIEW", but I still think it's worth applying the empty check for standard tables to avoid potential accidental data loss if the table already exists with data.  IMO if we want to allow overwriting existing populated tables we should have a `--force` option for that.

Reply:
I guess that changes slightly how initialization has been working. Most `init.sql` have `CREATE OR REPLACE TABLE` to explicitly overwrite existing tables. If we'd check for data then in most cases `CREATE OR REPLACE TABLE`  would effectively become `CREATE TABLE` (unless we are overwriting an empty table). 

I opened a follow-up issue: https://github.com/mozilla/bigquery-etl/issues/4798. Might be worth discussing in the workgroup meeting to see if people have a preference.

Reply:
Ah well, thinking about it having `--force` to overwrite existing tables does seem like a safer option. I'll add it to this PR